sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuApproximatePercentile.scala

```diff
@@ -0,0 +1,171 @@
+/*
+ * Copyright (c) 2021, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.nvidia.spark.rapids
+
+import com.nvidia.spark.rapids.GpuCast.{recursiveDoColumnar}
```

Comment:
I would prefer for us to keep the GpuCast in the call to recursiveDoColumnar otherwise it makes the code much harder to read and understand what is happening.

That or make the public facing API something like doCast.

Reply:
still want to see this happen.